src/index.ts

```diff
@@ -1584,7 +1583,25 @@ export class BigQuery extends common.Service {
       // The Job is important for the `queryAsStream_` method, so a new query
       // isn't created each time results are polled for.
       options = extend({job}, options);
-      job!.getQueryResults(options, callback as QueryRowsCallback);
+      if (
```

Comment:
Could you add a bit of context about why we're doing this extra logic? You could probably just reference the original performance issue & then mention why we need to treat the `[CREATE|REPLACE] MODEL` query types differently.

Reply:
Done